src/app/core/basic-datatypes/entity/entity.datatype.ts

```diff
@@ -46,17 +21,65 @@ export class EntityDatatype extends StringDatatype {
     super();
   }
 
-  override importMapFunction(
+  /**
+   * Maps a value from an import to an actual entity in the database by comparing the value with the given field of entities.
+   * Handles type conversion between numbers and strings to improve matching.
+   *
+   * @param val The value from an import that should be mapped to an entity reference
+   * @param schemaField The config defining details of the field that will hold the entity reference after mapping
+   * @param additional The field of the referenced entity that should be compared with the val
+   * @returns Promise resolving to the ID of the matched entity or undefined if no match is found
+   */
+  override async importMapFunction(
     val: any,
     schemaField: EntitySchemaField,
-    additional?: any,
-  ) {
-    if (!additional) {
-      return Promise.resolve(undefined);
+    additional?: string,
+  ): Promise<string | undefined> {
+    // If no additional field is specified or value is null/undefined, return undefined
+    if (!additional || val == null) {
+      return undefined;
+    }
+
+    // Convert value to string or number for flexible matching
+    const normalizedVal = this.normalizeValue(val);
+
+    try {
+      // Load all entities of the specified type
+      const entities = await this.entityMapper.loadType(schemaField.additional);
+
+      // Find the first entity where the specified field matches the normalized value
+      const matchedEntity = entities.find((entity) => {
+        const entityFieldValue = this.normalizeValue(entity[additional]);
+        return entityFieldValue === normalizedVal;
+      });
+
+      // Return the ID of the matched entity or undefined
+      return matchedEntity?.getId();
+    } catch (error) {
+      console.error("Error in EntityDatatype importMapFunction:", error);
```

Comment:
We have a special `Logging` service that we use instead of console.log in order to add some functionality like remote monitoring of errors. Please have a look here how to use that: https://aam-digital.github.io/ndb-core/documentation/additional-documentation/how-to-guides/log-errors.html

If you get stuck or have questions on this, do let us know.